### PR TITLE
Fix a bug that RealmSchema.remove() and RealmSchema.rename() don't update internal cache

### PR DESCRIPTION
**Pull request by zaki50**

### CODE REVIEW

File: realm/realm-library/src/main/java/io/realm/RealmSchema.java [hunk 1]
```diff
@@ -49,7 +49,7 @@
     // Caches Class objects (both model classes and proxy classes) to their Schema object
     private final Map<Class<? extends RealmModel>, RealmObjectSchema> classToSchema = new HashMap<>();
     // Caches Class Strings to their Schema object
-    private final Map<String, RealmObjectSchema> dynamicClassToSchema = new HashMap<>();
+    final Map<String, RealmObjectSchema> dynamicClassToSchema = new HashMap<>();
```
Comment (bmeike) [inline realm/realm-library/src/main/java/io/realm/RealmSchema.java]: If you are going to make this other than private, you need to control access to it.  Please wrap it in an `immutableHashMap`.

Comment (zaki50) [inline realm/realm-library/src/main/java/io/realm/RealmSchema.java]: But it's not immutable and actually modified from subclasses.

Comment (bmeike) [inline realm/realm-library/src/main/java/io/realm/RealmSchema.java]: I suggest, then, that you add `protected` methods to the superclass, that control the kinds of modifications that the subclass can perform.  A good rule is "`private` or `immutable`".  That is, even, in our lint config.

Comment (zaki50) [inline realm/realm-library/src/main/java/io/realm/RealmSchema.java]: @bmeike I've hidden the `dynamicClassToSchema` map again!

**Pull request merged**